locations/spiders/caseys_general_store.py

```diff
@@ -1,134 +1,28 @@
-                    opening_hours.add_range(
-                        day=day,
-                        open_time=open_time,
-                        close_time=close_time,
-                        time_format="%H:%M",
-                    )
-
-            elif operation_times[0] == "Mon-Fri":
-                for day in Seven_Days[:5]:
-                    opening_hours.add_range(
-                        day=day,
-                        open_time=open_time,
-                        close_time=close_time,
-                        time_format="%H:%M",
-                    )
-
-            elif operation_times[0] == "Sat-Sun":
-                for day in Seven_Days[5:]:
-                    opening_hours.add_range(
-                        day=day,
-                        open_time=open_time,
-                        close_time=close_time,
-                        time_format="%H:%M",
-                    )
-
-        return opening_hours.as_opening_hours()
+    item_attributes = {"brand": "Casey's General Store", "brand_wikidata": "Q2940968"}
+    allowed_domains = ["www.caseys.com"]
+    sitemap_urls = [
+        "https://www.caseys.com/sitemap.xml",
+    ]
+    sitemap_follow = [
+        r"https://www.caseys.com/medias/sys_master/root/.*/Store-en-USD-.*.xml",
+    ]
+    sitemap_rules = [
+        (
+            r"https://www.caseys.com/general-store/.*/[0-9]*",
+            "parse",
+        ),
+    ]
 
     def parse(self, response):
-        data = response.json()
-
-        for store in data["stores"]:
-            postal = store["address"]["postalCode"]
-            postal = postal.split("-")
-            amenities = store["amenities"]
-
-            properties = {
-                "ref": store["code"],
-                "name": store["displayName"],
-                "addr_full": store["address"]["line1"],
-                "city": store["address"]["town"],
-                "state": store["address"]["region"]["isocodeShort"],
-                "postcode": postal[0],
-                "country": store["address"]["region"]["countryIso"],
-                "lat": store["geoPoint"]["latitude"],
-                "lon": store["geoPoint"]["longitude"],
-                "phone": store["address"]["phone"],
-                "website": "https://www.caseys.com" + store["locationUrl"],
-                "extras": {
-                    "amenity:fuel": True,
-                    "car_wash": any("carwash" == a["key"] for a in amenities) or None,
-                    "fuel:diesel": any("diesel" == a["key"] for a in amenities) or None,
-                    "shop": "convenience",
-                },
-            }
+        item = LinkedDataParser.parse(response, "ConvenienceStore")
 
-            try:
-                h = self.parse_hours(store["storeOpenHours"])
-                if h:
-                    properties["opening_hours"] = h
-            except:
-                pass
+        item["ref"] = item["website"].split("/")[-1]
```

Comment:
We could also do a generic fallback ref = url [here](https://github.com/alltheplaces/alltheplaces/blob/a6d70ecd2afb0b1c6b235b12fa0a1d75fb3d0582/locations/linked_data_parser.py#L84) too

Reply:
Yeah, having a specific value is better, but I think we may end up with a bunch of spiders manually setting ref = url, if so handling it here would be easier. But we'll see, maybe most linked data will use branchCode or (at)id so it wont be necessary anyway.